spotinst/oceancd_verification_provider_datadog/fields_spotinst_oceancd_verification_provider_datadog.go

```diff
@@ -0,0 +1,139 @@
+package oceancd_verification_provider_datadog
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/service/oceancd"
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/commons"
+)
+
+func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
+
+	fieldsMap[DataDog] = commons.NewGenericField(
+		commons.OceanCDVerificationProviderDataDog,
+		DataDog,
+		&schema.Schema{
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					string(Address): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+
+					string(ApiKey): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+
+					string(AppKey): {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+				},
+			},
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			verificationProviderWrapper := resourceObject.(*commons.OceanCDVerificationProviderWrapper)
+			verificationProvider := verificationProviderWrapper.GetVerificationProvider()
+			var result []interface{} = nil
+
+			if verificationProvider != nil && verificationProvider.DataDog != nil {
+				result = flattenDataDog(verificationProvider.DataDog)
+			}
+			if len(result) > 0 {
+				if err := resourceData.Set(string(DataDog), result); err != nil {
+					return fmt.Errorf(string(commons.FailureFieldReadPattern), string(DataDog), err)
+				}
+			}
+			return nil
+		},
+
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			verificationProviderWrapper := resourceObject.(*commons.OceanCDVerificationProviderWrapper)
+			verificationProvider := verificationProviderWrapper.GetVerificationProvider()
+			var value *oceancd.DataDog = nil
+
+			if v, ok := resourceData.GetOk(string(DataDog)); ok {
+				if datadog, err := expandDataDog(v); err != nil {
+					return err
+				} else {
+					value = datadog
+				}
+			}
+			verificationProvider.SetDataDog(value)
+			return nil
+		},
+
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			verificationProviderWrapper := resourceObject.(*commons.OceanCDVerificationProviderWrapper)
+			verificationProvider := verificationProviderWrapper.GetVerificationProvider()
+			var value *oceancd.DataDog = nil
+
+			if v, ok := resourceData.GetOk(string(DataDog)); ok {
+				if datadog, err := expandDataDog(v); err != nil {
+					return err
+				} else {
+					value = datadog
+				}
+			}
+			verificationProvider.SetDataDog(value)
+			return nil
+		},
+		nil,
+	)
+}
+
+func expandDataDog(data interface{}) (*oceancd.DataDog, error) {
```

Comment:
expand to be re-checked, as all the attributes are required, we cannot set to nil